product_default_image/models/product_template.py

```diff
@@ -0,0 +1,78 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 LasLabs Inc.
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html).
+
+from odoo import api, fields, models, tools
+
+import logging
+_logger = logging.getLogger(__name__)
+
+
+class ProductTemplate(models.Model):
+
+    _inherit = 'product.template'
+
+    @api.model
+    def default_get(self, vals):
+        vals = super(ProductTemplate, self).default_get(vals)
+
+        if vals.get('categ_id'):
+
+            company = self.env.user.company_id
```

Comment:
So you're trying to return a specific company Record from `self.env.user.company_id`?

In your tests, you could do the following:
```
self.env.user.company_id = self.env['res.company'].browse(your_other_company_id)
```
It should be temporary, so you shouldn't have to reset the company_id before the next test fires off.